### DISCUSSION TITLE
求助自建服务后台http穿透问题

### DISCUSSION THREAD

**Discussion by stc15f104w** [natfrp/wiki#473] [使用问题] [answered]
rt，现在后台使用方案是http穿透，启动成功无报错，考虑短期调试避免备案走的美国的节点，配置好cname域名了，我这边能域名通到sakura的503页面，在老板那边的127.0.0.1：端口号能打开网页
这样情况求助可能的问题/可以试试的方案（tcp自动https？）
![image](https://github.com/natfrp/wiki/assets/36284558/8643c728-d82b-49f5-9152-468396a8e137)

thanks

[a comment by an automated account is not shown]

**Comment by fakeboboliu** [comment]
请多贴截图，否则只能算卦。

就端口来算的话，5001 是 dsm 用的 https 端口，可能需要使用 https 隧道。

**Comment by littledot2020** [comment]
有跨域问题吗？
我一个svelte站点，调用 https://frp-now.top:62711/api/config 时，
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://frp-now.top:62711/api/config. (Reason: CORS request did not succeed). Status code: (null).
我检查过后端python代码，已经做了跨域处理。帮我定位一下问题原因吧。

app = FastAPI(docs_url="/docs" if ENV == "dev" else None, redoc_url=None)
origins = ["*"]
app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

@app.get("/api/config")
async def get_app_config():
    # Checking and Handling the Absence of 'ui' in CONFIG_DATA

    default_locale = "zh-CN" # en-US
    if "ui" in CONFIG_DATA:
        log.info(f"ui in CONFIG_DATA===========")
        default_locale = CONFIG_DATA["ui"].get("default_locale", "zh-CN")   # en-US

    # The Rest of the Function Now Uses the Variables Defined Above
    return {
        "status": True,
        "name": WEBUI_NAME,
        "version": VERSION,
        "auth": WEBUI_AUTH,
        "default_locale": default_locale,
        "images": images_app.state.ENABLED,
        "default_models": webui_app.state.DEFAULT_MODELS,
        "default_prompt_suggestions": webui_app.state.DEFAULT_PROMPT_SUGGESTIONS,
        "trusted_header_auth": bool(webui_app.state.AUTH_TRUSTED_EMAIL_HEADER),
        "admin_export_enabled": ENABLE_ADMIN_EXPORT,
    }

**Reply by fakeboboliu** [reply]
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSDidNotSucceed
此处有你需要的信息

顺便善意提醒：下次问别人没有义务回答的问题前，先看看提问的艺术

**Comment by littledot2020** [comment]
报歉让你不开心了，向你致歉。上午我查找原因比较疲劳，所以说话没注意。向你道歉。
